articles/hapi-vs-koa-vs-express/index.md

```diff
@@ -0,0 +1,159 @@
+description: In this article we do a comparative analysis of three Node.js frameworks - we explore their advantages, disadvantages, and work an example to run a server in each.
+author: peter-kayere
+date: 2020-09-21T00:00:00-08:00
+topics: [Node.js]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/hapi-vs-koa-vs-express/hero.jpg
+    alt: koa koa.js Node.js express hapi framework
+---
+Node.js has various frameworks each of which is catered to the demands of users and developers. This article will give a comparison between the three major Node.js frameworks i.e. [express](https://expressjs.com), [hapi](https://hapi.dev) and [koa](https://koajs.com). In the comparison we will look at the difference between the three frameworks, their histories, advantages and disadvantages, popularity, support and performance.
+<!--more-->
+For more background on the different types of frameworks in Node.js, check out the following articles to get a more in depth look at Hapi, Koa, and Node.js before we begin:
+
+- [Introduction to Hapi.js framework](/engineering-education/introduction-to-hapi/)
+- [Introduction to Koa.js framework](/engineering-education/introduction-to-koajs/)
+- [Introduction to Node.js](/engineering-education/history-of-nodejs/)
+
+### Framework histories
+Before we start our in-depth comparison of each framework, lets take a look at a brief history on how each got started.
+
+#### Hapi.js
+Hapi.js was created by the mobile team at Walmart Labs — led by Eran Hammer to handle their traffic for events like Black Friday, which is by far one of the busiest days for online shopping on the U.S. calendar.
+Hapi's initial commit was made on August 5th, 2011 and after 5,619 commits later, it is currently at version 20.0.0. It was originally built using the express framework before facing challenges that drove Walmart to make hapi, its own stand-alone framework. Refer to my previous article on an [introduction to Hapi.js framework](/engineering-education/introduction-to-hapi/) for a detailed look at Hapi.js framework.
+
+#### Koa.js
+Koa's initial commit was made on August 17th, 2013 by TJ Holowaychuk. The framework was built by the team behind express with the aim of making a smaller, more expressive, and more robust framework. Koa.js is currently at version 2.13.0 with 1,086 commits. Koa.js is also known for having a smaller footprint compared to other frameworks. Refer to my previous article on an [introduction to koa.js framework](/engineering-education/introduction-to-koajs/) for a detailed look at Koa.js framework.
+
+#### Express.js
+Express framework is the oldest of the three. Its initial commit was made by TJ Holowaychuk on June 26th, 2009 and its first release was on the 2nd of January 2010 after around 660 commits. The two main contributors at the time of its first release were TJ Holowaychuk and Ciaron Jessup. Express.js is currently at version 4.17.1 after 280 releases and 5,592 commits.
```

Comment:
Add The at the start. You should add the reason why Express was developed to match the other two frameworks.